compiler/qsc_eval/src/lib.rs

```diff
@@ -1738,9 +1754,9 @@ fn eval_binop_shr(lhs_val: Value, rhs_val: Value) -> Value {
         Value::Int(val) => {
             let rhs = rhs_val.unwrap_int();
             if rhs > 0 {
-                Value::Int(val >> rhs)
+                Value::Int(val.wrapping_shr(rhs as u32))
             } else {
-                Value::Int(val << rhs.abs())
+                Value::Int(val.wrapping_shl(rhs.wrapping_abs() as u32))
             }
```

Comment:
I don't think that's a good option.  It's a rather peculiar detail that [`wrapping_shl`](https://dev-doc.rust-lang.org/beta/std/primitive.i64.html#method.wrapping_shl) is only defined for u32, even though there is [`shl(self, other: &i64)`](https://dev-doc.rust-lang.org/beta/std/primitive.i64.html#method.shl-86).  Assuming that the wrapping behavior is desired, we're only interested in the lower 6 bits of the rhs.  So one alternative that clippy would presumably be happier about would be `val << (rhs & 63)`, but I'd be less confident that would get optimized out, and it's less obvious at a glance that there's no way for that to panic, so I prefer using wrapping_shl to get this behavior.  Alternatively, if the wrapping behavior is not desired, we could use `checked_shl` and return Error::IntTooLarge in case of overflow.  But returning Error::IntTooLarge just because the rhs doesn't fit in a u32 would be completely arbitrary.

Reply:
Good point. In that case, I think `checked_shl` might be the right way to go. If users want to get a shift left that doesn't get mad about wrapping that's what `BigInt` can be used for. Personally, I think treating it as a runtime error if the user's chosen shift will cause "data loss" by dropping bits from the front is safer than wrapping here.

Reply:
Alright, I have changed the behavior to produce an error.